### PR TITLE
Model undeploy

### PR DESCRIPTION
**Pull request by jshum2479**
I have made the changes in both online and offline but no documentation yet.

### CODE REVIEW

File: core/src/main/python/wlsdeploy/tool/deploy/applications_deployer.py [hunk 1]
```diff
@@ -93,6 +96,14 @@ def __add_shared_libraries(self):
...
             self.logger.info('WLSDPLY-09608', LIBRARY, shared_library_name, self._parent_type, self._parent_name,
                              class_name=self._class_name, method_name=_method_name)
+
+            if model_helper.is_delete_name(shared_library_name):
+                location = LocationContext()
+                location.append_location(model_constants.LIBRARY)
+                existing_names = deployer_utils.get_existing_object_list(location, self.alias_helper)
+                deployer_utils.delete_named_element(location, shared_library_name, existing_names, self.alias_helper)
+                continue
+
```
[the remark this answers is not in the document]
Comment (rakillen) [reply core/src/main/python/wlsdeploy/tool/deploy/applications_deployer.py:106]: I don't think we can count on the app still being around in the archive, may have to examine the deployed one? ugh

[the remark this answers is not in the document]
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/tool/deploy/applications_deployer.py:106]: This is interesting question, do we allow user to specify a non-expanded name in the model for deletion?  Let say if the user has two versions of jsf shared library deployed. If the user only specify jsf which one should we delete ?

[the remark this answers is not in the document]
Comment (ddsharpe) [reply core/src/main/python/wlsdeploy/tool/deploy/applications_deployer.py:106]: Seems to me like the user should specify exactly what they want to delete.  If there are two versions of a library or app, delete by generic name would get too much.

**Comment by jshum2479** [comment]
I refactored the method for checking versioned app and error out if it is not the exact name, print out similar versioned name

**Pull request merged**